ui/screens.py

```diff
@@ -1403,6 +1416,200 @@ def poolClicked(self):
         """
         self.poolIp.setText(self.poolUrl.text())
 
+
+class AgendasScreen(Screen):
+    """
+    A screen that lists current agendas and allows for vote configuration.
+    """
+    def __init__(self, app, accountScreen):
+        """
+        Args:
+            app (TinyDecred): The TinyDecred application instance.
+        """
+        super().__init__(app)
+        self.isPoppable = True
+        self.canGoHome = True
+
+        self.pages = []
+        self.page = 0
+        self.ignoreVoteIndexChange = False
```

Comment:
do you mean to set the vote?

Reply:
This property appears to be used to prevent concurrent calls to `onChooseChoice`. Am I reading that right? 

This is okay, though I question the utility. Were you seeing some undesirable behavior without it?

Reply:
There is an onconnected() function listening to changes in the dropdowns/comboboxes. It is triggered when a user clicks a new choice, which is good. It also triggers when we initially set the vote, which is not optimal.

For instance we make the dropdown and default index is zero. After checking with the user's pool their choice is "yes" at index 1, so we change it. Because we change it the onconnected () fires and sends an unnecessary request to the pool to set the vote to yes, although it is already yes.

Another way to avoid this is to ignore changes until we are synced, but I made this with the idea that a user can have different vote choices for different pools.  So the problem happens again upon switching pools. I could change it so that the user has one one choice per wallet, and then we just set everything to that choice and set all new, old pools to the same vote choice, then I don't think this variable would be necessary.

Unless you have a better way?

Reply:
Instead of `currentIndexChanged`, connect to the `activated` signal. The only caveat is that you'll have to verify something has actually changed before proceeding, because `activated` will be triggered even if they select the same thing that was already selected.

Reply:
Ok, I'll do it that way. Thanks!

Reply:
ok. I think this is optimized,  but let me know.